### PR TITLE
[FLOC-3263] Delete all Docker containers as part of cluster cleanup.

### PR DESCRIPTION
**Pull request by itamarst**
Various changes to make Docker plugin tests pass more:

1. Cleanup Docker containers before cluster is configured. Otherwise a container may still be running with a volume attached, preventing volume from being deleted in cleanup and therefore making test fail.
2. Deleted bogus use of find_free_port (the fact port is free on machine running acceptance tests is no indication of whether it'll be free on cluster nodes!).
3. Increase timeout on test that is often quite slow on Rackspace, and therefore times out even though it's still making slow progress.

### CODE REVIEW

File: flocker/acceptance/endtoend/test_dockerplugin.py [hunk 1]
```diff
@@ -160,7 +163,7 @@ def _test_create_container(self, cluster, volume_name=None):
         node = cluster.nodes[0]
         client = get_docker_client(cluster, node.public_address)
         http_port = 8080
-        host_port = find_free_port()[1]
+        host_port = 8080
```
Comment (exarkun) [inline flocker/acceptance/endtoend/test_dockerplugin.py]: I fixed this properly in another branch.  Could you drop these changes so I don't have a conflict to resolve?

**Comment by jml** [comment]
No other suggestions. Please amend as per @exarkun's suggestion & merge when tests are OK.

**Comment by wallrj** [comment]
Sorry @itamarst 

I started this yesterday and didn't finish it.

Only thing I notice is that when I tried to run trial again against the `--keep`ed nodes I get the following error:

```
8c1fccf0-e185-4477-98bf-ef8d59089688 -> /4385
2015-12-09 02:30:08.469066Z
  message_type: flocker.common.runner:run:stdout
  line: 2015-12-08 18:28:27-0800 [HTTP11ClientProtocol (TLSMemoryBIOProtocol),client] ELIOT: {"task_uuid": "5441b7fe-5a85-4f49-a2a4-631da1bb340f", "result": "True", "action_type": "flocker:common:loop_until", "timestamp": 1449628107.430079, "action_status": "succeeded", "task_level": [3, 3, 3]}

8c1fccf0-e185-4477-98bf-ef8d59089688 -> /4386
2015-12-09 02:30:08.469346Z
  message_type: flocker.common.runner:run:stdout
  line: 2015-12-08 18:28:27-0800 [HTTP11ClientProtocol (TLSMemoryBIOProtocol),client] ELIOT: {"timestamp": 1449628107.430384, "task_uuid": "5441b7fe-5a85-4f49-a2a4-631da1bb340f", "action_type": "acceptance:cleanup_containers", "action_status": "succeeded", "task_level": [3, 4]}

8c1fccf0-e185-4477-98bf-ef8d59089688 -> /4387
2015-12-09 02:30:08.469694Z
  message_type: flocker.common.runner:run:stdout
  line: 2015-12-08 18:30:07-0800 [HTTP11ClientProtocol (TLSMemoryBIOProtocol),client] ELIOT: {"exception": "requests.exceptions.ConnectTimeout", "timestamp": 1449628207.469629, "task_uuid": "5441b7fe-5a85-4f49-a2a4-631da1bb340f", "action_status": "failed", "reason": "HTTPSConnectionPool(host='10.229.84.209', port=2376): Max retries exceeded with url: /v1.21/containers/json?all=0&limit=-1&trunc_cmd=0&size=0 (Caused by ConnectTimeoutError(<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7ff86cc980d0>, 'Connection to 10.229.84.209 timed out. (connect timeout=100)'))", "action_type": "acceptance:clean_nodes", "errno": null, "task_level": [4]}


```

It might be a quirk of my set up though. 

The initial `./admin/run-acceptance-tests` failed because it was pulling old packages from buildbot.

I rebuilt the forced  buildbot to rebuild the master packages and installed them on the nodes using  the `--provider managed` managed option.

And the test failure is because the cleanup_all_containers function attempts to connect to the private IP address rather than the public.

Please merge if you can get green build results.

**Comment by itamarst** [comment]
The private IP thing doesn't happen on Jenkins. And it's an unrelated preexisting issue, I assume, insofar as that cleanup code was there already, I just moved it.

**Pull request merged**